File: src/types/prompt.ts
export interface PromptVersionDto {
  id: string;
  promptId: string;
  version: number;
  promptContent: string;
  createdAt: string;
}

export interface DefaultPromptDto {
  id: string;
  promptKey: string;
  name: string;
  description: string | null;
  activeVersionId: string;
  activeVersion: PromptVersionDto;
  createdAt: string;
  updatedAt: string;
}

File: src-tauri/src/domain/prompt.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Prompt {
    pub id: String,
    pub prompt_key: String,
    pub name: String,
    pub description: Option<String>,
    pub active_version_id: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct PromptVersion {
    pub id: String,
    pub prompt_id: String,
    pub version: i64,
    pub prompt_content: String,
    pub created_at: String,
}
